src/main/java/com/purbon/search/fair/utils/BinarySearchAlphaAdjuster.java

```diff
@@ -21,123 +29,76 @@ public BinarySearchAlphaAdjuster(int n, int k, double p, double alpha) {
-            if (succProb <= 0.00001 && succProb > 0) {
+            if (Math.abs(succProb - alpha) <= 0.0001) {
                 return adjustedAlpha;
+            } else if (Math.abs(succProb - alpha) < Math.abs(succProb - minOptAlpha)) {
+                minOptAlpha = adjustedAlpha;
             }
-            if (0.00001 < succProb) {
+            if (alpha < succProb) {
                 right = adjustedAlpha - STEP;
             } else {
                 left = adjustedAlpha + STEP;
             }
-            if (succProb < min) {
-                secondMin = min;
-                secondMinOptAlpha = minOptAlpha;
-                min = succProb;
-                minOptAlpha = adjustedAlpha;
-
-            }
 
         }
-        return secondSearch(minOptAlpha, secondMinOptAlpha);
-    }
-
-    private double secondSearch(double left, double right) {
-        double step = 0.00000001;
-        AlphaAdjuster adj = new AlphaAdjuster(k, p, left);
-        double oldSuccProb = 1 - adj.computeSuccessProbability();
-        ArrayList<SuccessProbAlphaPair> values = new ArrayList<>();
-        while (left < right) {
-            left = left + step;
-            AlphaAdjuster adjuster = new AlphaAdjuster(k, p, left);
-            double succProb = 1 - adjuster.computeSuccessProbability();
-            if (oldSuccProb < succProb) {
-                break;
-            }
-            oldSuccProb = succProb;
-            values.add(new SuccessProbAlphaPair(succProb, left));
-        }
-        return values.size() > 1 ? getMinAlpha(values) : left;
-    }
-
-    private double getMinAlpha(ArrayList<SuccessProbAlphaPair> list) {
-        double min = list.get(0).getSuccProb();
-        double alpha = list.get(0).getAlpha();
-        for (SuccessProbAlphaPair p : list) {
-            if (p.getSuccProb() < min) {
-                min = p.getSuccProb();
-                alpha = p.getAlpha();
-            }
-        }
-        return alpha;
-    }
-
-    public int getN() {
-        return n;
-    }
-
-    public void setN(int n) {
-        this.n = n;
+        return minOptAlpha;
     }
 
     public int getK() {
         return k;
     }
 
-    public void setK(int k) {
-        this.k = k;
-    }
-
     public double getP() {
         return p;
     }
 
-    public void setP(double p) {
-        this.p = p;
-    }
-
     public double getAlpha() {
         return alpha;
     }
 
-    public void setAlpha(double alpha) {
-        this.alpha = alpha;
-    }
-
-    private class SuccessProbAlphaPair {
-        double succProb;
-        double alpha;
-
-        private SuccessProbAlphaPair(double succProb, double alpha) {
-            this.succProb = succProb;
-            this.alpha = alpha;
-        }
-
-        private double getSuccProb() {
-            return succProb;
-        }
-
-        private double getAlpha() {
-            return alpha;
+    public double getAdjustedAlpha() {
+        if (adjustedAlpha == 0.0) {
```

Comment:
you should not need this case, actually this function does two things, what is not a good practise 😄